pkg/messaging/direct_messaging.go

```diff
@@ -123,6 +126,8 @@ func (d *directMessaging) invokeWithRetry(
 		if err == nil {
 			return resp, nil
 		}
+		log.Errorf("retry count: %d, grpc call failed, ns: %s, addr: %s, appid: %s, err: %s",
```

Comment:
Can you change this to `log.Debugf`? Since this is a per-call operation it might pollute the log output.